flink-java/src/main/java/org/apache/flink/api/java/utils/RequiredParameter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Facility to manage required parameters in user defined functions.
+ */
+public class RequiredParameter {
+
+	private static final String HELP_TEXT_PARAM_DELIMITER = "\t";
+	private static final String HELP_TEXT_LINE_DELIMITER = "\n";
+
+	private HashMap<String, Option> data;
+
+	public RequiredParameter() {
+		this.data = new HashMap<>();
+	}
+
+	public void add(Option option) throws RequiredParameterException {
```

Comment:
We could add an additional method, though then we do have an implicit coupling between that add method and the constructor of Option. If, and that is probably a big if, the constructor were to change, we would have to update the add method as well. 

Only providing a String based version of the add method would lead to code like that:
```
Option option = req.add("name")
option.type(...).values(...)
```

which in my opinion is a little bit awkward compared to instantiation, setting the right  type and values and then adding the object.

what do you think?